### PR TITLE
CORE-9974: add key parameter when creating secrets

### PR DESCRIPTION
**Pull request by dickon**
Some secrets implementations need to store keys in an external system, and that requires a stable key to match with the value. Consider for instance a simple secrets implementation which is a dictionary; the implementation needs to be given the key to associate with the value it gets.

So, add a key parameter, and pass it in everywhere, trying to arrange for keys to be globally unique. I'm not sure we'll get this precisely right on the first attempt and it isn't clear how to automatically test uniquness of secret keys.

### CODE REVIEW

File: components/virtual-node/virtual-node-write-service-impl/src/main/kotlin/net/corda/virtualnode/write/db/impl/writer/VirtualNodeDbFactory.kt [hunk 1]
```diff
@@ -154,7 +154,8 @@ class VirtualNodeDbFactory(
             val config = createDbConfig(
                 smartConfigFactory, user, password.concatToString(),
                 jdbcUrl = jdbcUrl,
-                maxPoolSize = maxPoolSize
+                maxPoolSize = maxPoolSize,
+                key = "virtual-node-database-password"
```
Comment (driessamyn) [inline components/virtual-node/virtual-node-write-service-impl/src/main/kotlin/net/corda/virtualnode/write/db/impl/writer/VirtualNodeDbFactory.kt]: This key needs to be unique for each vault DB, doesn't it?

I'd suggest something like `corda-vault-$holdingIdentityShortHash-password`

File: tools/plugins/initial-config/src/main/kotlin/net/corda/cli/plugin/initialconfig/CryptoConfigSubcommand.kt [hunk 2]
```diff
@@ -81,8 +81,11 @@ class CryptoConfigSubcommand : Runnable {
         } else {
             softHsmRootSalt!!
         })
-        val wrappingPassphraseSecret = smartConfigFactory.makeSecret(wrappingPassphraseDefined).toSafeConfig().root()
-        val wrappingSaltSecret = smartConfigFactory.makeSecret(wrappingSaltDefined).toSafeConfig().root()
+        val wrappingPassphraseSecret =
+            smartConfigFactory.makeSecret(wrappingPassphraseDefined, "master-wrapping-key-passphrase").toSafeConfig()
+                .root()
+        val wrappingSaltSecret =
+            smartConfigFactory.makeSecret(wrappingSaltDefined, "master-wrapping-key-salt").toSafeConfig().root()
```
Comment (driessamyn) [inline tools/plugins/initial-config/src/main/kotlin/net/corda/cli/plugin/initialconfig/CryptoConfigSubcommand.kt]: Should we prefix the keys with `corda` maybe?

File: tools/plugins/initial-config/src/main/kotlin/net/corda/cli/plugin/initialconfig/DbConfigSubcommand.kt [hunk 3]
```diff
@@ -112,22 +112,24 @@ class DbConfigSubcommand : Runnable {
...
 
+@Suppress("LongParameterList")
 fun createDbConfig(
     jdbcUrl: String,
     username: String,
     password: String,
     jdbcPoolMaxSize: Int,
-    secretsService: SecretsCreateService
+    secretsService: SecretsCreateService,
+    key: String = "database-password"
```
Comment (driessamyn) [inline tools/plugins/initial-config/src/main/kotlin/net/corda/cli/plugin/initialconfig/DbConfigSubcommand.kt]: We shouldn't default this, it doesn't make sense.

**Review by driessamyn — CHANGES_REQUESTED** [review]

File: components/virtual-node/virtual-node-write-service-impl/src/main/kotlin/net/corda/virtualnode/write/db/impl/writer/VirtualNodeDbFactory.kt (hunk 1 above)
Comment (dickon) [reply components/virtual-node/virtual-node-write-service-impl/src/main/kotlin/net/corda/virtualnode/write/db/impl/writer/VirtualNodeDbFactory.kt]: fixed

File: tools/plugins/initial-config/src/main/kotlin/net/corda/cli/plugin/initialconfig/CryptoConfigSubcommand.kt (hunk 2 above)
Comment (dickon) [reply tools/plugins/initial-config/src/main/kotlin/net/corda/cli/plugin/initialconfig/CryptoConfigSubcommand.kt]: Why not, good idea. My expectation is that people would use a dedicated corda secret in vault, but that's not guaranteed and we don't know in this code if there are separate secret each with their own key space.

File: tools/plugins/initial-config/src/main/kotlin/net/corda/cli/plugin/initialconfig/DbConfigSubcommand.kt (hunk 3 above)
Comment (dickon) [reply tools/plugins/initial-config/src/main/kotlin/net/corda/cli/plugin/initialconfig/DbConfigSubcommand.kt]: Déjà vu.

defualt removed.

**Review by driessamyn — APPROVED** [review]
LGTM

**Pull request merged**